gokart/target.py

```diff
@@ -16,9 +16,17 @@
 from gokart.zip_client_util import make_zip_client
 from gokart.object_storage import ObjectStorage
 
+from gokart.redis_lock import RedisParams, make_redis_params, with_lock
+
 logger = getLogger(__name__)
 
 
+def _decorate_file_access_functions_with_redis_lock(super, self, redis_params):
+    self.load = with_lock(func=super.load, redis_params=redis_params)
```

Comment:
@mski-iksm 

I think it better adding lock at function definition
https://github.com/m3dev/gokart/blob/61d59a02fa42fcd17fc607f4d081659efdcef8d5/gokart/target.py#L35
than replacing the method from outside of the class.

```python
class TargetOnKart(luigi.Target):
    @with_lock
    def load(self) -> Any:
        return self._load()
```

Reply:
Or it can be clear if you write with_lock as TargetOnKart's method 

```python
class TargetOnKart(luigi.Target):
    def _with_lock(self, func):
        return with_lock(func, self.redis_params)

    def load(self) -> Any:
        return self._with_lock(self._load())
```